assets/dev/js/frontend/scripts-import-manager.js

```diff
@@ -0,0 +1,53 @@
+export default class ScriptsImportManager extends elementorModules.ViewModule {
+	getRegisteredModuleScripts() {
+		return [
+			{
+				moduleKey: 'carousel-base',
+				moduleName: 'CarouselBase',
+				frontendObject: 'frontendHandlers',
+				importFunction: () => import( /* webpackChunkName: 'carouselBaseHandler' */ './handlers/base-carousel' ),
+			},
+		];
+	}
+
+	getActiveModuleScripts() {
+		return ! elementorFrontend.isEditMode() && !! elementorScriptsImports ? elementorScriptsImports : [];
+	}
+
+	getFrontendObject( objectName ) {
+		let frontendObject = null;
+
+		switch ( objectName ) {
+			case 'frontendHandlers':
+				frontendObject = elementorModules.frontend.handlers;
+				break;
+			case 'utils':
+				frontendObject = elementorFrontend.utils;
+				break;
+		}
+
+		return frontendObject;
+	}
+
+	onInit() {
+		this.registeredModuleScripts = this.getRegisteredModuleScripts();
+		this.activeModuleScripts = this.getActiveModuleScripts();
+		this.loadActiveModuleScripts();
+	}
+
+	loadActiveModuleScripts() {
+		// TODO: Remove this in version 3.28 [ED-15983].
```

Comment:
@hein-obox Please clarify what part should be removed, the const or the entire function.

Reply:
@rami-elementor 
We should keep the default loading for at least two versions, so that we can update the community about this, and so that we can handle Core/Pro dependencies in a safe way.
After two versions we can remove the parameter and implement the conditional loading.